### ISSUE TITLE
[key packages] High Impact Packages

### ISSUE THREAD

**Issue by Eomm** [nodejs/package-maintenance#143]
Split 2/3 of #105

We have to to choose a larger group of high impact packages to implement the same practices and standards developed in conjunction with the Pilot Packages.

To choose this group of packages, we need to define the right formula to find who need help first.

The starting list of criteria for ranking is:
➕ = higher value increment the priority assigned
➖  = lower value increment the priority assigned

| Criteria | Evaluation for priority |
|----------|------------|
| Number of downloads | ➕ |
| Number of downloads of their full dependency tree✨ | ➕ |
| Most depended-upon packages | ➕ |
| Number of dependencies | ➖ |
| Test against Node.js versions | ➕, less is to seek and understand why |
| Old dependencies | Less are to seek and understand why. "old" is managable or do we have many false positive? |
| Deprecated dependencies | ➖ |
| Open issues | ➕ - only if labelled as `confirmed bug` or `bug` |
| Is it maintained by a company? | They want help? |
| Last activities | More recent: ➕ |

✨ During this process, we should also check if one of those dependency is on an older major and why.

## What to discuss here

+ We need to complete the criteria list and define the evaluation of the criteria itself

**Comment by ljharb** [comment]
What makes a dependency “old”?

**Comment by Eomm** [comment]
I think that a module could be considered old when there is a newer version (at least minor/patch) or it has dependencies module deprecated.
And, as @dominykas points out, "seek and understand why" could fit better: maybe the updated lib doesn't support old versions of node.

I wonder if need more help who have dependencies updated or not 🤔

**Comment by TheHollidayInn** [comment]
For "Open issues" - is higher better because it means it is active?

**Comment by mhdawson** [comment]
Once we have general agreement I think getting the list that would be the top 10 for these would be a good way to validate them.

**Comment by Eomm** [comment]
@ljharb Yeah, maybe I have understood what you say. Could you suggest criteria to understand these cases? Otherwise, I would treat that as "false positive". I really don't know how many could be these case 😅 I trust of your experience.

@TheHollidayInn I think that high usage of the module and high open bugs means a high level of help needed, but you pointed a good question: should we also consider if the package is "active"? 
ex: less time has passed from the last commit is better

**Comment by ljharb** [comment]
I think almost every criteria we come up with is likely to be a false positive, and false positives are a very dangerous thing for "best practices" we want to evangelize.

Certainly getting a concrete list based on the latest list of criteria is a good start to evaluating them.

**Comment by mcollina** [comment]
I would remove the term "better" as it might create bad feelings in maintainers.

**Comment by tomByrer** [comment]
May I suggest adding how frequent a project is downloaded via a CDN, such as [jsDelivr](https://www.jsdelivr.com/)?  Several packages there are downloaded over a billion times per month, but jsDelivr has [several layers of caching](https://www.jsdelivr.com/network) so npm.com will very rarely see those hits.

**Comment by Eomm** [comment]
Nice question, because till now we had evaluated only the official npm registry.

I need to study more the context because my first thought is that it could be out of scope.
We are in trying to help the modules that run in node.js environment, but the modules download via cdn live in the browser for what I know (and a fast search it seems that jsDelivr doesn't expose a registry).
Tell me if I'm wrong

**Comment by tomByrer** [comment]
Thanks for your consideration @Eomm .
JsDelivr used to use GitHub as a file repo, but was unmanageable (eg ate half of a local HD).
It duplicates npm & GitHub, not sure automatically or on-demand, but I assume "Load any project hosted on npm" means all one way or another.

npm is used for front-end projects often; see how many hits jQuery & Bootstrap have?  Granted those are well maintained now, but likely there are some with 10_000+ hits/mo that might skew your numbers.  The [98th ranking](https://data.jsdelivr.com/v1/stats/packages/month), Foundation, had 454k hits last month.

jsDelivr has big clients listed on their front page. [cdnjs](https://trends.builtwith.com/cdn/CDN-JS) many more users, & [unpkg.com](https://trends.builtwith.com/cdn/UNPKG) is getting some ranking also.
